tf2_gnn/data/ppi_dataset.py

```diff
@@ -151,33 +130,37 @@ def __load_data(self, data_dir: RichPath, data_fold: DataFold) -> List[PPIGraphS
             graph_node_offset = graph_id_to_node_offset[graph_id]
             src_node, tgt_node = src_node - graph_node_offset, tgt_node - graph_node_offset
 
-            cur_graph_data = graph_id_to_graph_data[graph_id]
-            cur_graph_data.adjacency_lists[fwd_edge_type].append((src_node, tgt_node))
-            cur_graph_data.type_to_node_to_num_inedges[fwd_edge_type][tgt_node] += 1
-            if not self._params["tie_fwd_bkwd_edges"]:
-                cur_graph_data.adjacency_lists[bkwd_edge_type].append((tgt_node, src_node))
-                cur_graph_data.type_to_node_to_num_inedges[bkwd_edge_type][src_node] += 1
-            else:
-                cur_graph_data.adjacency_lists[fwd_edge_type].append((tgt_node, src_node))
-                cur_graph_data.type_to_node_to_num_inedges[fwd_edge_type][src_node] += 1
+            graph_id_to_edges[graph_id].append((src_node, tgt_node))
 
         final_graphs = []
-        for graph_data in graph_id_to_graph_data.values():
-            # numpy-ize:
-            adj_lists = []
-            for edge_type_idx in range(self.num_edge_types):
-                adj_lists.append(np.array(graph_data.adjacency_lists[edge_type_idx]))
+        for graph_id in graph_id_to_edges.keys():
+            num_nodes = len(graph_id_to_features[graph_id])
+
+            adjacency_lists, type_to_node_to_num_inedges = self._process_raw_adjacency_lists(
+                raw_adjacency_lists=[graph_id_to_edges[graph_id]], num_nodes=num_nodes
+            )
+
             final_graphs.append(
                 PPIGraphSample(
-                    adjacency_lists=adj_lists,
-                    type_to_node_to_num_inedges=np.array(graph_data.type_to_node_to_num_inedges),
-                    node_features=np.array(graph_data.node_features),
-                    node_labels=np.array(graph_data.node_labels),
+                    adjacency_lists=adjacency_lists,
+                    type_to_node_to_num_inedges=type_to_node_to_num_inedges,
+                    node_features=np.array(graph_id_to_features[graph_id]),
+                    node_labels=np.array(graph_id_to_labels[graph_id]),
                 )
             )
 
         return final_graphs
 
+    def _process_raw_adjacency_lists(
```

Comment:
It seems that this is only called in line 139, so I would tend to inline it there?

Reply:
Yes, I intentionally extracted `_process_raw_adjacency_lists` to be the same in all three datasets, so that it's more unified and easier to refactor further. I think it's used only once in `QM9Dataset` as well. Would you recommend to inline it here and leave the other two?